### PR TITLE
Remove unmerged closed pull requests from milestones

### PR DESCRIPTION
**Pull request by yardenshoham**
Milestone maintenance will make sure any unmerged pull request will not be part of any milestone.

### CODE REVIEW

**Comment by yardenshoham** [comment]
From local testing: 
image

File: src/github.ts [hunk 1]
```diff
@@ -66,6 +66,21 @@ export const fetchTargeting = async (branch: string) => {
...
+) => {
+  const response = await fetch(
+    `${GITHUB_API}/search/issues?q=` +
+      encodeURIComponent(
+        `is:pr is:closed is:unmerged milestone:${milestoneTitle} repo:go-gitea/gitea`,
+      ),
+    { headers: HEADERS },
+  );
+  const json = await response.json();
+  return json;
```
Comment (silverwind) [inline src/github.ts:81]: Does this API return `[]` when no matches?

Comment (yardenshoham) [reply src/github.ts:81]: Yes

Comment (silverwind) [reply src/github.ts:81]: Ah, what luxure when one has a well-behaved API to talk to 😆

Comment (silverwind) [reply src/github.ts:81]: (For some reason, I can not resolve comments in this repo)

Comment (yardenshoham) [reply src/github.ts:81]: GitHub's API is pretty good IMO

**Review by silverwind — APPROVED** [review]

File: src/milestones.ts [hunk 2]
```diff
@@ -47,3 +47,30 @@ export const assign = async (prNumber: number) => {
...
 };
+
+export const run = async () => {
+  await removeMilestonesFromUnmergedClosedPrs();
+};
+
+// removes milestones from all PRs that are closed and unmerged
+const removeMilestonesFromUnmergedClosedPrs = async () => {
+  // get the current open milestones
+  const milestones = await github.getMilestones();
```
Comment (lunny) [inline src/milestones.ts:58]: We just need to get open milestones. Closed milestones could be ignored.

Comment (yardenshoham) [reply src/milestones.ts:58]: Yes, `github.getMilestones()` returns the open milestones

**Review by lunny — APPROVED** [review]

**Pull request merged**